runtime/src/iree/hal/utils/deferred_work_queue.c

```diff
@@ -913,7 +1151,7 @@ iree_hal_deferred_work_queue_execution_device_signal_host_callback(
   IREE_TRACE_ZONE_BEGIN(z0);
   iree_hal_deferred_work_queue_action_t* action =
       (iree_hal_deferred_work_queue_action_t*)user_data;
-  IREE_ASSERT_EQ(action->kind, IREE_HAL_QUEUE_ACTION_TYPE_EXECUTION);
+  IREE_ASSERT_LE(action->kind, IREE_HAL_QUEUE_ACTION_TYPE_QUEUE_DEALLOC);
```

Comment:
if doing these checks you want to add an `IREE_HAL_QUEUE_ACTION_TYPE_MAX = ` at the end of the enum and set it to the last field. any place caring about the max can check that and updates to the enum are local (new enum value just updates MAX at the end)